test/test_compile.py

```diff
@@ -48,8 +45,12 @@ def _test_compile_base(
     x = torch.randn(*x_shape, device="cuda", dtype=linear_dtype)
     m_ref = nn.Linear(16, 32, bias=True, device="cuda", dtype=linear_dtype)
 
-    m_fp8 = get_float8_linear(
-        linear_type, m_ref, emulate, scaling_type_x, scaling_type_w, scaling_type_dL_dY
+    m_fp8 = Float8Linear.from_float(
```

Comment:
calling 'swap_..' on nn.Linear module returns a model out of place. I think its fine either way

Reply:
I agree, we can make the tests use that if we want in a future PR.